### ISSUE TITLE
custom local rule : srcip : ipv6 :  Invalid ip address validation

### ISSUE THREAD

**Issue by aderumier** [wazuh/wazuh#14699]
version : wazuh 4.3.6  - debian 11

Hi, I'm trying to create this custom rule to exclude an ipv6 srcip

```
  <rule id="100102" level="0">
    <if_level>1</if_level>
    <srcip>2a0a:1580::a</srcip>
    <description>Ignoring all alerts from this ipv6 ip</description>
  </rule>
```

When I'm trying to validate, I got:

```
Error file content is incorrect: Could not check validation (1908) - Error validating configuration: (1237): Invalid ip address: '2a0a:1580::a'., (1220): Error loading the rules: 'etc/rules/local_rules.xml'.
Could not check validation (1908) - Error validating configuration: (1237): Invalid ip address: '2a0a:1580::a'., (1220): Error loading the rules: 'etc/rules/local_rules.xml'.
Error: Could not check validation (1908) - Error validating configuration: (1237): Invalid ip address: '2a0a:1580::a'., (1220): Error loading the rules: 'etc/rules/local_rules.xml'. 
    at Function.returnErrorInstance (https://server.domain.com:5601/1/bundles/plugin/wazuh/wazuh.plugin.js:1:117968)
    at Function._callee2$ (https://server.domain.com:5601/1/bundles/plugin/wazuh/wazuh.plugin.js:1:116188)
    at tryCatch (https://server.domain.com:5601/1/bundles/plugin/indexManagementDashboards/indexManagementDashboards.plugin.js:1:94916)
    at Generator.invoke [as _invoke] (https://server.domain.com:5601/1/bundles/plugin/indexManagementDashboards/indexManagementDashboards.plugin.js:1:98867)
    at Generator.next (https://server.domain.com:5601/1/bundles/plugin/indexManagementDashboards/indexManagementDashboards.plugin.js:1:96055)
    at asyncGeneratorStep (https://server.domain.com:5601/1/bundles/plugin/wazuh/wazuh.plugin.js:1:109980)
    at _next (https://server.domain.com:5601/1/bundles/plugin/wazuh/wazuh.plugin.js:1:110291)
```

Same rule works fine with an ipv4.

**Comment by aderumier** [comment]
same with manually modify the file and reload wazuh-manager

```
août 19 17:04:57 server.domain.com: env[21688]: 2022/08/19 17:04:57 wazuh-analysisd: ERROR: (1237): Invalid ip address: '2a0a:1580::a'.
août 19 17:04:57 server.domain.com: env[21688]: 2022/08/19 17:04:57 wazuh-analysisd: CRITICAL: (1220): Error loading the rules: 'etc/rules/local_rules.xml'.
```

**Comment by jnasselle** [comment]
Hi @aderumier,

Thanks for reporting this issue. A fix was introduced in #10892 and will be part of the next Wazuh minor version (v4.4).

In the meantime to workaround this, a [CDB list](https://documentation.wazuh.com/current/user-manual/ruleset/cdb-list.html) (containing the IPv6 address) could be used instead:

- Add list to `ossec.conf` ([ref](https://documentation.wazuh.com/current/user-manual/reference/ossec-conf/ruleset.html#list))
```xml
  <list>etc/lists/ipv6</list>
```
- Create the list (`etc/lists/ipv6`) file with permissions 660 and ownership wazuh:wazuh
```
"2a0a:1580::a":
```
- Modify your custom rule
```xml
  <rule id="100102" level="0">
    <if_level>1</if_level>
    <list field="srcip" lookup="match_key">etc/lists/ipv6</list>
    <description>Ignoring all alerts from this ipv6 ip</description>
  </rule>
```
After restarting the Wazuh Manager this could be tested with `wazuh-logtest` with a sample log

```
root@wazuh-dev:/var/ossec/etc/lists# /var/ossec/bin/wazuh-logtest
Starting wazuh-logtest v4.3.7
Type one log per line

date=2016-06-16 time=10:19:23 devname=Device_Name devid=FGTXXXX9999999999 logid=0000000013 type=traffic subtype=forward level=notice vd=root srcip=2a0a:1580::a srcport=57727 srcintf="internal1" dstip=7.8.9.81 dstport=80 dstintf="wan1" poluuid=d6217c58-8c42-51e5-c3a6-c7766895cbfd sessionid=181876 proto=6 action=deny policyid=8 dstcountry="Reserved" srccountry="Reserved" trandisp=snat transip=160.242.8.82 transport=57727 service="HTTP" appid=107347980 app="Proxy.HTTP" appcat="Proxy" apprisk=critical applist="default" appact=drop-session duration=30 sentbyte=0 rcvdbyte=3042 sentpkt=0 utmaction=block countapp=1 crscore=10 craction=1048576

**Phase 1: Completed pre-decoding.
        full event: 'date=2016-06-16 time=10:19:23 devname=Device_Name devid=FGTXXXX9999999999 logid=0000000013 type=traffic subtype=forward level=notice vd=root srcip=2a0a:1580::a srcport=57727 srcintf="internal1" dstip=7.8.9.81 dstport=80 dstintf="wan1" poluuid=d6217c58-8c42-51e5-c3a6-c7766895cbfd sessionid=181876 proto=6 action=deny policyid=8 dstcountry="Reserved" srccountry="Reserved" trandisp=snat transip=160.242.8.82 transport=57727 service="HTTP" appid=107347980 app="Proxy.HTTP" appcat="Proxy" apprisk=critical applist="default" appact=drop-session duration=30 sentbyte=0 rcvdbyte=3042 sentpkt=0 utmaction=block countapp=1 crscore=10 craction=1048576'

**Phase 2: Completed decoding.
        name: 'fortigate-firewall-v5'
        action: 'deny'
        app: 'Proxy.HTTP'
        appact: 'drop-session'
        appcat: 'Proxy'
        appid: '107347980'
        applist: 'default'
        apprisk: 'critical'
        countapp: '1'
        craction: '1048576'
        crscore: '10'
        devid: 'FGTXXXX9999999999'
        devname: 'Device_Name'
        dstcountry: 'Reserved'
        dstintf: 'wan1'
        dstip: '7.8.9.81'
        dstport: '80'
        duration: '30'
        level: 'notice'
        logid: '0000000013'
        policyid: '8'
        poluuid: 'd6217c58-8c42-51e5-c3a6-c7766895cbfd'
        proto: '6'
        rcvdbyte: '3042'
        sentbyte: '0'
        sentpkt: '0'
        service: 'HTTP'
        sessionid: '181876'
        srccountry: 'Reserved'
        srcintf: 'internal1'
        srcip: '2a0a:1580::a'
        srcport: '57727'
        subtype: 'forward'
        time: '10:19:23'
        trandisp: 'snat'
        transip: '160.242.8.82'
        transport: '57727'
        type: 'traffic'
        utmaction: 'block'
        vd: 'root'

**Phase 3: Completed filtering (rules).
        id: '100102'
        level: '0'
        description: 'Ignoring all alerts from this ipv6 ip'
        groups: '['local', 'syslog', 'sshd']'
        firedtimes: '1'
        mail: 'False'
```
Hope this helps

**Comment by jnasselle** [comment]
Hi @aderumier ,

I am closing this issue due to inactivity. I assume your question was solved. Feel free to open it again if need anything else.

Regards
